aptos-move/framework/aptos-framework/sources/vesting_without_staking.move

```diff
@@ -439,19 +445,24 @@ module aptos_framework::vesting_without_staking {
                 amount: shareholder_amount,
             },
         );
+
+        // remove `shareholder_address`` from `vesting_contract.shareholders`
         let shareholders = &mut vesting_contract.shareholders;
-        simple_map::remove(shareholders, &shareholder_address);
-        let shareholder_baneficiary = *simple_map::borrow(&vesting_contract.beneficiaries, &shareholder_address);
-        simple_map::remove(&mut vesting_contract.beneficiaries, &shareholder_baneficiary);
-        assert!(simple_map::contains_key(shareholders, &shareholder_address), 0);
+        assert!(simple_map::contains_key(shareholders, &shareholder_address), error::not_found(ESHAREHOLDER_NOT_EXIST));
+        let (_, shareholders_vesting) = simple_map::remove(shareholders, &shareholder_address);
+
+        // remove `shareholder_address` from `vesting_contract.beneficiaries`
+        let shareholder_beneficiaries = &mut vesting_contract.beneficiaries;
+        assert!(simple_map::contains_key(shareholder_beneficiaries, &shareholder_address), error::not_found((ESHAREHOLDER_NOT_EXIST)));
```

Comment:
This assertion need not hold since not all `shareholders` may have `beneficiaries`. So if a beneficiary exist for a shareholder, then only it should be removed.

Reply:
> This assertion need not hold since not all `shareholders` may have `beneficiaries`. So if a beneficiary exist for a shareholder, then only it should be removed.

@sjoshisupra Based on the implementation of set_beneficiary, I think all the shareholders have beneficiaries. It could be the same with the shareholder address.